### DISCUSSION TITLE
Meeting import: confused

### DISCUSSION THREAD

**Discussion by OSAwebsite** [code4recovery/12-step-meeting-list#570] [Q&A] [answered]
Hi team,

I have an issue with importing from a CSV file. It is not so much a discussion topic, but a bug... I think...

I had not used that import feature for awhile, but since the formatting and syntax are preprogrammed in my external datasource, I used the existing logic.

I found the following unexpected behaviour:
Everything seems to be imported properly, but the group information (in the bottom of the back end of the listing) seems to be set to the default value "Individual meeting" and not accept group contact information anymore. When I toggle that value to "Part of a group", the information I expected to see on the website is not displayed, even though it was part of the CSV file.

That information is essential for my listings not in the least because it contains the official unique WSO ID for each group.

In the example sheet for import I don't see anything referring to this contact information setting. I think it used to be that if there was a value in the 'group' section of the CSV file, it would automatically toggle to 'Part of a group' and display what I need to be displayed.

This is now no longer the case.
There is also no datasource in the plugin that I can fill myself with group contact information.
I do not keep personal details on file for groups, but I do use the group notes section for valuable information.

From what I can see during my testing, the _group name_ **IS** preserved somewhere in the backend of the plugin. Because when I prepared a new upload for the same group, and still did not get what I wanted, I entered the group name manually and it was recognised by the plugin. However, the related information (District and group notes) no longer showed up.

I use the import functionality in the plugin for bulk imports. I expect one to be coming when the vast majority of my meetings is in-person again. I then will need to adjust the group information in my external datasource and upload it to the website via the plugin. In my external datasource that is done within 5 minutes. It would be counter productive to do the bulk import and then still have to browse through each individual meeting (I have well over 300!), using the painfully slow and wonky Wordpress interface. 

So please: HELP!

I can see one of two changes, if your team is willing to look into this:

1. the toggle is automatically set to 'part of group', when there is a value in group name (not to be confused with the meeting name, which is often different from the group name)

or

2. the toggle is integrated in the import procedure and can be declared by me, so that the interface understands to set it and 
thus can display information properly on the front end.

For your reference, I am adding the CSV and 3 screenshots for your reference. If there are any questions, please let me know.

Oh and before I forget: I am the website coordinator for an **Al-Anon Area website** with hundreds of meetings.
When I mention the unique ID for groups, I am referring to our WSO ID - I assume in AA you have the same thing that might be referred to as GSO ID. That value is still sorely missing from the plugin, by the way. I have asked to include it in its own box many times, to no avail. Which is why I add it to the group notes so new GRs and other members in service are able to find it in their meeting listings and use it in all communications about their groups to others in service and to our World Service Office (WSO).

Because I am not an AA website, I do not have to worry about keeping information in sync with the AA mobile meeting app. Which is why I have no problem with deleting all meeting content and refreshing it with a bulk upload. I often adjust individual meetings manually in Wordpress, but if there are many, that is just a major pain in the you know what...

Thanks for your consideration.

**Import file in csv format:**

[LTL2.csv](https://github.com/code4recovery/12-step-meeting-list/files/7530114/LTL2.csv)

**Incomplete listing:**

![Incomplete listing upload](https://user-images.githubusercontent.com/59859095/141527863-40920198-e76c-4b4c-ac51-5d0e94727a7c.jpg)

**Correct listing:**

![Correct listing after manual toggle](https://user-images.githubusercontent.com/59859095/141527865-3ba3df7c-426b-40cf-89da-3d4cf0bfc81e.jpg)

**Location in the back end of the group contact information toggle:**

![Group Contact Information Section with toggle](https://user-images.githubusercontent.com/59859095/141527967-01935ba7-ce51-4689-a750-ac8688560101.jpg)

**Comment by brianw-area24** [comment, marked as answer]
Thank you for including the example import file, it does help when we have information like this because we can see better what's going on. 

When I opened that file, it looks like some of the information is in the incorrect columns.
* Sub District contains "Group ID: 30769637, District 22. Meeting type: Al-Anon. Focus: Regular. Open to Families and Friends of alcoholics only." I'm thinking this column should be empty
* Group Notes contains "Families and Friends Only,English", I'm thinking it should contain the information that's in the Sub District Column
* Types is empty, but should contain "Families and Friends Only,English"
* 
After correcting the columns, it appears to import correctly on my test site.
![Screen Shot 2021-12-07 at 11 54 54 PM](https://user-images.githubusercontent.com/55821195/145156200-9d3569f7-28c4-4405-88d9-90f66413095d.png)
